src/NextjsDistribution.ts

```diff
@@ -448,7 +392,7 @@ export class NextjsDistribution extends Construct {
       ...cfDistributionProps,
 
       // these values can NOT be overwritten by cfDistributionProps
-      domainNames,
+      domainNames: this.props.domainNames,
       certificate: this.certificate,
```

Comment:
This value is coming from an uninitiated instance variable (as far as I can tell!).

`this.certificate,` needs to be created or passed in from the Domain construct.